### PR TITLE
Add private project switch action

### LINKED ISSUE #125: Add the PrivateProjectSwitchTapped action
## Description
This action is emitted when the user taps the row in the project view that controls whether the project is private or not.

## Dependencies
Depends on #111 

## Definition of done
- [x] The Action is added
- [x] The action is dispatched when you tap the row in the project view
- [x] When the action is handled, it toggles the property
- [x] The UI should reflect the change
- [x] Tests ensure behavior is correct
- [x] The documentation is updated to reflect the changes done in this PR

### PR DESCRIPTION
**Pull request by juanlaube**
## Info
Adds the private project switch action

### Summary
Adds the private project switch action and handle that in the reducer.

### Technical considerations
Nothing special, just an action that is sent from a button.

### Relationships
Closes #125 

### Pull Request checklist
- [x] The code does not hinder the app's performance
- [x] The code does not introduce regressions
- [x] The code in this PR is tested thoroughly enough that changing the behavior to an incorrect one breaks tests

- [ ] The code used custom templates

### Review template

- [ ] The code is thoroughly tested in well written and resilient manner
- [ ] The code is performant and leads to an optimal user experience
- [ ] The code is testable and does not cut corners for the sake of delivering
- [ ] The code considers older platforms well and will work in older devices just like it will in new ones

- [ ] The code considers light and dark theme
- [ ] The code considers RTL and does not mention left/right
- [ ] The code considers i18n
  - [ ] There are no hardcoded strings in the UI
  - [ ] All strings are included in every language we support OR there's an issue requesting new translations
- [ ] There are no hard-coded numbers referring to dimensions or font sizes
- [ ] The code considers UI elements that are similar and extracts their similarities into styles to avoid duplication

### CODE REVIEW

File: Modules/Timer/Timer/Project/Views/ProjectViewController.swift [hunk 1]
```diff
@@ -43,5 +44,16 @@ public class ProjectViewController: UIViewController, Storyboarded {
             .map(ProjectAction.nameEntered)
             .bind(onNext: store.dispatch)
             .disposed(by: disposeBag)
+
+        // Private project
+        store.select({ $0.editableProject })
+            .map({ $0?.isPrivate })
+            .drive(onNext: { self.togglePrivateProjectButton.isHighlighted = $0 ?? false })
```
Comment (Odrakir) [inline Modules/Timer/Timer/Project/Views/ProjectViewController.swift:51]: I think it's cleaner to create a selector that filters nil and returns just a bool

File: Modules/Timer/Timer/Project/ProjectSelector.swift [hunk 2]
```diff
@@ -1 +1,5 @@
 import Models
+
+let editableProjectSelector: (ProjectState) -> EditableProject = { state in
+    return state.editableProject!
```
Comment (sofokli1) [inline Modules/Timer/Timer/Project/ProjectSelector.swift:4]: is there always a value?

Comment (Odrakir) [reply Modules/Timer/Timer/Project/ProjectSelector.swift:4]: I'm with @sofokli1 . How about adding a new method on store called `compactSelect` for example?

```swift
func compactSelect<B>(_ selector: @escaping (State) -> B?) -> Driver<B> 
func compactSelect<B>(_ selector: @escaping (State) -> B?) -> Driver<B> where B: Equatable
```

Comment (juanlaube) [reply Modules/Timer/Timer/Project/ProjectSelector.swift]: Take a look again please.

File: Modules/Utils/Utils/RxExtensions/Driver+Extensions.swift [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import Foundation
+import RxSwift
+import RxCocoa
+
+extension Driver {
+    public func compactMap<B>(_ transform: @escaping (Element) -> B?) -> Driver<B> {
+        return map(transform).filter({ $0 != nil }).asDriver(onErrorJustReturn: nil).map({ $0! })
```
Comment (Odrakir) [inline Modules/Utils/Utils/RxExtensions/Driver+Extensions.swift:7]: Nice!

File: Modules/Timer/Timer/Project/Views/ProjectViewController.swift [hunk 4]
```diff
@@ -34,14 +34,26 @@ public class ProjectViewController: UIViewController, Storyboarded {
     public override func viewDidLoad() {
         super.viewDidLoad()
 
-        store.select({ $0.editableProject })
-            .map({ $0?.name })
+        // Project name
+        store.compactSelect({ $0.editableProject })
+            .map({ $0.name })
```
Comment (Odrakir) [inline Modules/Timer/Timer/Project/Views/ProjectViewController.swift:39]: I think this reads better
```suggestion
        store.compactSelect({ $0.editableProject?.name })
```

File: Modules/Timer/Timer/Project/Views/ProjectViewController.swift [hunk 5]
```diff
@@ -34,14 +34,26 @@ public class ProjectViewController: UIViewController, Storyboarded {
...
             .disposed(by: disposeBag)
 
         projectNameTextField.rx.text.compactMap({ $0 })
             .map(ProjectAction.nameEntered)
             .bind(onNext: store.dispatch)
             .disposed(by: disposeBag)
+
+        // Private project
+        store.compactSelect({ $0.editableProject })
+            .map({ $0.isPrivate })
```
Comment (Odrakir) [inline Modules/Timer/Timer/Project/Views/ProjectViewController.swift:50]:
```suggestion
        store.compactSelect({ $0.editableProject?.isPrivate })
```

**Pull request merged**